source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.md

```diff
@@ -0,0 +1,1501 @@
+
+The unstructured form SHOULD be as follows, using the values defined in the structured format above to fill in
+placeholders as appropriate:
+
+> Connection pool for {{serverHost}}:{{serverPort}} cleared for serviceId {{serviceId}}
+
+#### Pool Closed Message
+
+In addition to the common fields defined above, this message MUST contain the following key-value pairs:
+
+| Key     | Suggested Type | Value                    |
+| ------- | -------------- | ------------------------ |
+| message | String         | "Connection pool closed" |
+
+The unstructured form SHOULD be as follows, using the values defined in the structured format above to fill in
+placeholders as appropriate:
+
+> Connection pool closed for {{serverHost}}:{{serverPort}}
+
+#### Connection Created Message
+
+In addition to the common fields defined above, this message MUST contain the following key-value pairs:
+
+| Key                | Suggested Type | Value                                                                               |
+| ------------------ | -------------- | ----------------------------------------------------------------------------------- |
+| message            | String         | "Connection created"                                                                |
+| driverConnectionId | Int64          | The driver-generated ID for the connection as defined in [Connection](#connection). |
+
+The unstructured form SHOULD be as follows, using the values defined in the structured format above to fill in
+placeholders as appropriate:
+
+> Connection created: address={{serverHost}}:{{serverPort}}, driver-generated ID={{driverConnectionId}}
+
+#### Connection Ready Message
+
+In addition to the common fields defined above, this message MUST contain the following key-value pairs:
+
+| Key                | Suggested Type | Value                                                                               |
+| ------------------ | -------------- | ----------------------------------------------------------------------------------- |
+| message            | String         | "Connection ready"                                                                  |
+| driverConnectionId | Int64          | The driver-generated ID for the connection as defined in [Connection](#connection). |
+| durationMS         | Int64          | `ConnectionReadyEvent.duration` converted to milliseconds.                          |
+
+The unstructured form SHOULD be as follows, using the values defined in the structured format above to fill in
+placeholders as appropriate:
+
+> Connection ready: address={{serverHost}}:{{serverPort}}, driver-generated ID={{driverConnectionId}}, established
+> in={{durationMS}} ms
+
+#### Connection Closed Message
+
+In addition to the common fields defined above, this message MUST contain the following key-value pairs:
```

Comment:
Some of the tables, but not all of them, have been converted to html tables.  Is this something we'd want to fix (maybe there's an unsupported feature we're using in the tables, so they need to be html)?  If so, I can suggest the changes to convert them to markdown tables.

Reply:
Markdown doesn't support multi-line entries in the table cells.  If we want to make them into Markdown tables we'll have to collapse the entries into a single line.

Reply:
I'm fine either way and can make the changes, do you have a preference?

Reply:
Okay, I'll go for a MD table instead of an HTML table.